platform/broadcom/sonic-platform-modules-quanta/ix9-32x/sonic_platform/sfp.py

```diff
@@ -321,56 +69,22 @@ def __init__(self, sfp_index, sfp_type):
         }
 
         for x in range(self.PORT_START, self.PORT_END + 1):
-            port_eeprom_path = eeprom_path.format(self.port_to_i2c_mapping[x])
+            port_eeprom_path = sfp_eeprom_path.format(self.port_to_i2c_mapping[x])
             self.port_to_eeprom_mapping[x] = port_eeprom_path
 
-        self._detect_sfp_type(sfp_type)
-
-        self.info_dict_keys = ['type', 'hardware_rev', 'serial', 'manufacturer',
-            'model', 'connector', 'encoding', 'ext_identifier',
-            'ext_rateselect_compliance', 'cable_type', 'cable_length',
-            'nominal_bit_rate', 'specification_compliance', 'vendor_date', 'vendor_oui', 'application_advertisement']
-
-        SfpBase.__init__(self)
-
-
-    def _convert_string_to_num(self, value_str):
-        if "-inf" in value_str:
-            return 'N/A'
-        elif "Unknown" in value_str:
-            return 'N/A'
-        elif 'dBm' in value_str:
-            t_str = value_str.rstrip('dBm')
-            return float(t_str)
-        elif 'mA' in value_str:
-            t_str = value_str.rstrip('mA')
-            return float(t_str)
-        elif 'C' in value_str:
-            t_str = value_str.rstrip('C')
-            return float(t_str)
-        elif 'Volts' in value_str:
-            t_str = value_str.rstrip('Volts')
-            return float(t_str)
-        else:
-            return 'N/A'
+        SfpOptoeBase.__init__(self)
 
-    def __read_txt_file(self, file_path):
-        try:
-            with open(file_path, 'r') as fd:
-                data = fd.read()
-                return data.strip()
-        except IOError:
-            pass
-        return ""
+    def get_port_type(self):
+        """
+        Retrieves the port/cage type of this SFP
 
-    def __is_host(self):
-        return os.system(self.HOST_CHK_CMD) == 0
+        Returns:
+            A string, the port/cage type of this SFP
+        """
+        return self.SFP_PORT_TYPE_QSFPDD
```

Comment:
Shouldn't this (at least ideally) be based on byte 00 from the plugged in module?

Reply:
This is the cage type of the port, not the media type fetched from the EEPROM of the attached transceiver.

Reply:
ack, thanks.